### PR TITLE
Added feature for deleting user reports and related data by either id, title, or zipped files

### PR DESCRIPTION
**Pull request by priyansh1913**
## Description

This PR addresses sub-issue #215 of the parent feature Issue #214. It implements the function to delete a user report and all related data from the database, supporting deletion by report title, id, zipped file path.

**Just a Note** - The database schema was updated to add the required title and zipped_filepath columns to the user report table, as these fields were not present previously but are necessary for the requested functionality.

**Closes:** #215 

---

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation added/updated
- [X] Test added/updated
- [ ] Refactoring
- [ ] Performance improvement

---

## Testing

- [X] Test deletion by user report title
- [X] Test deletion by zipped file path (direct and from preferences)
- [X]  Test deletion removes related project and file report rows
- [X] Test referential integrity and FK constraints are maintained
- [X] Test deletion when multiple reports share the same project/file
- [X] Test deletion with empty or non-existent title/filepath
- [X]  Test edge cases: undated/untitled reports, similar values, and empty database

---

## Checklist

- [X] GenAI was used in generating the code and I have performed a self-review of my own code
- [X] I have commented my code where needed
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## Screenshots

> If applicable, add screenshots to help explain your changes

<details>
<summary>Click to expand screenshots</summary>

</details>

### CODE REVIEW

File: src/database/utils/database_modify.py [hunk 1]
```diff
@@ -75,3 +76,38 @@ def create_row(report: FileReport | ProjectReport | UserReport):
         if hasattr(row, col_name):
             setattr(row, col_name, value)
     return row
+
+
+def delete_user_report_and_related_data(report_id=None, title=None, zipped_filepath=None):
```
[the remark this answers is not in the document]
Comment (priyansh1913) [reply src/database/utils/database_modify.py]: So just to clarify, do you mean that my function should return the list of userReports after the deletion by title

[the remark this answers is not in the document]
Comment (priyansh1913) [reply src/database/utils/database_modify.py]: Yes that makes sense, it would be easier to have it in another PR and merge that to develop

**Review by jademola — APPROVED** [review]
Looks good to go. If you can either make a PR or issue which address the need for user's to add a title to their reports would be great. Thanks, Priyansh!

File: src/database/utils/database_modify.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alextaschuk) [reply src/database/utils/database_modify.py]: I like this idea Jimi. I just realized that when I made the issue for this PR, I said that the user should be able to specify which user report they want to delete by giving it a name (which Priyansh has since added), but there isn't any logic in the `UserReport` class to store or use the report's name. I think asking the user to name the report after it is generated is a good way to do this, and if they don't provide one, it wil default to the name of the zipped folder.

File: src/database/db.py [hunk 2]
```diff
@@ -137,6 +137,8 @@ class UserReportTable(Base):
     __tablename__ = 'user_report'
 
     id = mapped_column(Integer, primary_key=True)
+    title = mapped_column(String)
+    zipped_filepath = mapped_column(String)
```
[the remark this answers is not in the document]
Comment (alextaschuk) [reply src/database/db.py]: Yeah, I'll take myself off of that issue. I think that we'll need to create a new issue for adding logic to prompt the user to enter a name for a user report after we print the results. The logic should include updating the row in the `user_report` table too.

**Review by alextaschuk — APPROVED** [review]
Thanks Priyansh, this all looks great. I left a couple of replies under Jimi's comments if you could take a look.

**Review by alextaschuk — CHANGES_REQUESTED** [review]

File: src/database/utils/database_modify.py [hunk 3]
```diff
@@ -75,3 +76,38 @@ def create_row(report: FileReport | ProjectReport | UserReport):
...
+    """
+    Delete a user report and all related project and file reports by id, title, or zipped_filepath.
+    Args:
```
[the remark this answers is not in the document]
Comment (priyansh1913) [reply src/database/utils/database_modify.py]: I have now changed the function with the try-except statement.
